src/main/groovy/edu/oregonstate/mist/api/Resource.groovy

```diff
@@ -136,16 +136,18 @@ abstract class Resource {
      * The path is copied from the request path.
      * The query parameters are taken from the params argument.
      *
-     * Falsey params are omitted from the url.
+     * Falsely params are omitted from the url.
      * The parameters pageNumber and pageSize are
      * converted to page[number] and page[size].
      *
      * @param params a map of query parameters for the url
+     * @param resourceEndpoint: the endpoint to be appended on the uri.
      * @return the url
      */
-    protected String getPaginationUrl(Map params) {
-        URIBuilder uriBuilder = new URIBuilder(endpointUri).setPath(uriInfo.requestUri.path)
+    protected String getPaginationUrl(Map params, String resourceEndpoint) {
+        String path = endpointUri.path + resourceEndpoint
```

Comment:
Is there a method you can use to join paths? Let's not assume that endpointUri.path ends with a slash, or that resourceEndpoint starts with one. Or if we are going to assume that, at least document it somewhere.